src/SprykerEco/Zed/AvalaraTax/Business/Mapper/AvalaraTransactionRequestMapper.php

```diff
@@ -0,0 +1,192 @@
+
+    /**
+     * @param \SprykerEco\Zed\AvalaraTax\AvalaraTaxConfig $avalaraTaxConfig
+     * @param \SprykerEco\Zed\AvalaraTax\Dependency\Facade\AvalaraTaxToMoneyFacadeInterface $moneyFacade
+     * @param \SprykerEco\Zed\AvalaraTaxExtension\Dependency\Plugin\CreateTransactionRequestExpanderPluginInterface[] $createTransactionRequestExpanderPluginInterfaces
+     */
+    public function __construct(
+        AvalaraTaxConfig $avalaraTaxConfig,
+        AvalaraTaxToMoneyFacadeInterface $moneyFacade,
+        array $createTransactionRequestExpanderPluginInterfaces
+    ) {
+        $this->avalaraTaxConfig = $avalaraTaxConfig;
+        $this->moneyFacade = $moneyFacade;
+        $this->createTransactionRequestExpanderPluginInterfaces = $createTransactionRequestExpanderPluginInterfaces;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\CalculableObjectTransfer $calculableObjectTransfer
+     * @param \Generated\Shared\Transfer\AvalaraCreateTransactionRequestTransfer $avalaraCreateTransactionRequestTransfer
+     *
+     * @return \Generated\Shared\Transfer\AvalaraCreateTransactionRequestTransfer
+     */
+    public function mapCalculableObjectTransferToAvalaraCreateTransactionRequestTransfer(
+        CalculableObjectTransfer $calculableObjectTransfer,
+        AvalaraCreateTransactionRequestTransfer $avalaraCreateTransactionRequestTransfer
+    ): AvalaraCreateTransactionRequestTransfer {
+        $avalaraCreateTransactionTransfer = $this->createAvalaraCreateTransactionTransfer($calculableObjectTransfer);
+
+        $isTaxIncluded = $this->isTaxIncluded($calculableObjectTransfer);
+        foreach ($calculableObjectTransfer->getItems() as $itemTransfer) {
+            $avalaraLineItemTransfer = $this->mapItemTransferToAvalaraLineTransfer($itemTransfer, new AvalaraLineItemTransfer());
+            $avalaraLineItemTransfer->setTaxIncluded($isTaxIncluded);
+            $avalaraCreateTransactionTransfer->addLine($avalaraLineItemTransfer);
+        }
+
+        return $this->executeCreateTransactionRequestExpanderPlugins(
+            $calculableObjectTransfer,
+            $avalaraCreateTransactionRequestTransfer->setTransaction($avalaraCreateTransactionTransfer)
+        );
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\CalculableObjectTransfer $calculableObjectTransfer
+     *
+     * @return \Generated\Shared\Transfer\AvalaraCreateTransactionTransfer
+     */
+    protected function createAvalaraCreateTransactionTransfer(CalculableObjectTransfer $calculableObjectTransfer): AvalaraCreateTransactionTransfer
+    {
+        $avalaraCreateTransactionTransfer = (new AvalaraCreateTransactionTransfer())
+            ->setCurrencyCode($calculableObjectTransfer->getCurrencyOrFail()->getCodeOrFail())
+            ->setCompanyCode($this->avalaraTaxConfig->getCompanyCode())
+            ->setWithCommit($this->avalaraTaxConfig->getIsTransactionCommitAfterOrderPlacementEnabled());
```

Comment:
* It's is only applicable when `CalculableObjectTransfer` has an order reference.

Reply:
Resolved.